test/test_mock.py

```diff
@@ -1591,6 +1613,11 @@ def a_coroutine(fut):
     # fine as long as the implementation is shared with a generator. Also, it's
     # really hard to fall in a case like this one with a coroutine.
     def test_patch_stopped_when_generator_is_collected(self):
+        if platform.python_implementation() == 'PyPy':
```

Comment:
Please use the skip decorators:

    @unittest.skipIf(sys.platform != "CPython", "Test relying on how __del__ is called by implementation")

and add a test similar to this one which uses gen.close() instead of del gen, for all implementations.

I think we also need a test which ensures that patchings are applied when gen.close() is called, something like:

    def gen():
        try:
            while True:
                yield None
        except GeneratorExit:
            # test that patches are still active here
            return